### DISCUSSION TITLE
information on making new smart contracts

### DISCUSSION THREAD

**Discussion by NoobML** [Cyfrin/foundry-full-course-cu#549] [Q&A]
//SPDX-License-Identifier:MIT

pragma solidity 0.8.18;

import{SimpleStorage} from"./SimpleStorage.sol";

contract StorageFactory
{
   SimpleStorage[] public listOfSimpleStorageContracts;

 function createSimpleStorageContract() public 
 {
    SimpleStorage newSimpleStoragecontract = new SimpleStorage();
    // why can't i wrtite like this for making new contracts 
    // SimpleStorage newSimpleStoragecontract;
    // listOfSimpleStorageContracts.push(newSimpleStoragecontract);
    // instead of adding assignment opertator and then adding a "new" keyowrd followed by the contract name again 
    listOfSimpleStorageContracts.push(newSimpleStoragecontract);
 }

**Comment by Naman1729** [comment]
Hey @Mushiiiiii ,
```solidity
SimpleStorage newSimpleStoragecontract = new SimpleStorage();
``` 
This line of code creates a new instance of the SimpleStorage contract and assigns it to the newSimpleStoragecontract variable. This is the correct way to create a new contract instance in Solidity. you cannot create a new contract instance without using the **new** keyword followed by the contract's name. Solidity requires the explicit use of the **new** keyword to create new contract instances.

```solidity
SimpleStorage newSimpleStoragecontract;
```
It will result in a compilation error, as you are not providing a valid initialization for the newSimpleStoragecontract variable. it does not contain the instance of contract so you can not able to call function by this variable.
